Source/WebCore/dom/Document.cpp

```diff
@@ -2988,7 +2988,8 @@ void Document::collectRangeDataFromRegister(Vector<WeakPtr<HighlightRangeData>>&
 {
     for (auto& highlight : highlightRegister.map()) {
         for (auto& rangeData : highlight.value->rangesData()) {
-            if (rangeData->startPosition().isNotNull() && rangeData->endPosition().isNotNull())
+            // FIXME: Change to a flag that checks if liveRange was changed rather than if liveRange for optimization.
```

Comment:
```suggestion
            // FIXME: For live ranges, we can optimize by only performing this when the range changed.
```

It took me a bit to understand the current comment

Reply:
Will change!